Add rspec and some specs

emsk opened the pull request:
Usage: `rake`

Results:
```
Rhymer::CLI
  #spit
    given short text
      should output "今日は良い天気 こんな日は自然に元気\n" to stdout
    given long text
      should output /25日大坂城に入城 将軍は大阪を発ち同月16日入京\n/ to stdout

Rhymer::Lyric
  .new
    given text
      should be an instance of Rhymer::Lyric
      should respond to #lyric
      should not respond to #lyric=
    given nil text
      should raise ArgumentError with "Text to parse cannot be nil"
  #lyric
    should include (an instance of Natto::MeCabNode)
  #mecab
    should be an instance of Natto::MeCab
  #nauns
    should include {0 => (an instance of Natto::MeCabNode), 4 => (an instance of Natto::MeCabNode)}
    #[0]
      #feature
        should match /\A名詞/

Rhymer::Parser
  .new
    given text
      should be an instance of Rhymer::Parser
      should respond to #lyric
      should not respond to #lyric=
      should respond to #rhymes
      should not respond to #rhymes=
    given nil text
      should raise NoMethodError with "undefined method `gsub' for nil:NilClass"
  #lyric
    should be an instance of Rhymer::Lyric
  #rhymes
    with default config
      given rhymed text
        should eq [["今日は良い天気", "こんな日は自然に元気", 20]]
      given no rhymed text
        should eq []
    with custom config
      given low config[:vibes_threshold]
        should eq [["今日は良い天気", "こんな日は自然に元気", 20]]
      given high config[:vibes_threshold]
        should eq []
      given config[:prefix_length]
        should eq [["良い天気", "こんな日は自然に元気", 20]]

Rhymer
  VERSION
    should not be nil

Finished in 1.61 seconds (files took 0.39487 seconds to load)
23 examples, 0 failures
```

I decided to use RSpec because spec/spec_helper.rb already exists.

現状（v0.0.4）の実装に合わせてテストコードを書いてみました。
既に spec/spec_helper.rb がありましたので、テスティングフレームワークには RSpec を使用しています。

suzuki86 commented:
いつもありがとうございます。確認します！

suzuki86 commented on spec/rhymer/cli_spec.rb:
```diff
@@ -0,0 +1,26 @@
+      end
+
+      it { is_expected.to output(/25日大坂城に入城 将軍は大阪を発ち同月16日入京\n/).to_stdout }
+    end
```
Wikipedia の文章が Gem に含まれるとライセンス上、好ましくない可能性がありますので
念のためこの `context` は削除頂けますでしょうか？

https://github.com/emsk/rhymer/blob/10b88247c3b505d582284b34105ae69c52d429d5/spec/rhymer/cli_spec.rb#L12-L24

emsk commented:
Thank you for reviewing.
I removed the text and updated related spec code a little.

Result:
```
Rhymer::CLI
  #spit
    given text
      should output "今日は良い天気 こんな日は自然に元気\n" to stdout
```

ご確認、ご指摘いただきありがとうございます。
当該箇所を削除しました。
また、これによって冗長になった同ファイル内の spec も少し修正をしておりますので、再度ご確認いただければと思います 🙏

suzuki86 commented:
修正ありがとうございます！
最後に、Wikipedia の文章を含んだコミットを消すために
コミットのスカッシュをお願いできますか？

suzuki86 commented:
ありがとうございます :pray:
